pkg/workloads/docker.go

```diff
@@ -563,89 +558,3 @@ func (d *dockerClient) GetAllInfraContainersPID() (map[string]int, error) {
 
 	return pids, nil
 }
-
-// finishIpvlanInit finishes configuring ipvlan slave device of the given endpoint.
-//
-// Unfortunately, Docker libnetwork itself moves a netdev to netns of a container
-// after the Cilium libnetwork plugin driver has responded to a `JoinEndpoint`
-// request. During the move, the netdev qdisc's get flushed by the kernel. Therefore,
-// we need to configure the ipvlan slave device in two stages.
-//
-// Because the function can be called many times for the same container in parallel,
-// we need to make the function idempotent. This is achieved by checking
-// whether the datapath map has been pinned, which indicates previous
-// successful invocation of the function for the same container, before executing
-// the configuration stages.
-//
-// FIXME: Because of the libnetwork limitation mentioned above, we cannot enforce
-// policies for an ipvlan slave before a process of a container has started. So,
-// this enables a window between the two stages during which ALL container traffic
-// is allowed.
-func finishIpvlanInit(ep *endpoint.Endpoint, netNsPath string) error {
-	var ipvlanIface string
-
-	if netNsPath == "" {
```

Comment:
We lost this nil-check. Are we now guaranteed that it's non-nil?

Reply:
Good catch! Will be fixed in next upload.